test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java

```diff
@@ -1682,6 +1682,28 @@ public void test83_verifier(RunInfo info) {
         }
     }
 
+    /*
+    TODO: 8335256: Properly handle merging of value object oops
+    @Test
+    @IR(failOn = {CALL_UNSAFE, ALLOC})
+    public MyValue1 test84(MyValue1 v) {
+        v = U.makePrivateBuffer(v);
+        for (int i = 0; i < 10; i++) {
+            U.putInt(v, X_OFFSET, i);
+        }
+        U.putInt(v, X_OFFSET, rI);
```

Comment:
Here, crash occurs becausethe compiler expects to receive an InlineTypeNode for  value type objects, since we are now buffering this value in make private buffer hence we hit following assertion failure

https://github.com/openjdk/valhalla/blob/lworld/src/hotspot/share/opto/parse1.cpp#L1789

This test does not capture a scenario where we merge a larval and non-larval value.
https://github.com/user-attachments/files/19501609/unsafe_access.txt